components/script/dom/element.rs

```diff
@@ -2108,6 +2111,13 @@ impl ElementMethods for Element {
             None => return Err(Error::NotSupported)
         }
     }
+
+    #[allow(unrooted_must_root)]
+    fn RequestFullscreen(&self) -> Rc<Promise> {
+        let doc = document_from_node(self);
+        doc.set_fullscreen_element(Some(self));
+        Promise::new(self.global().r())
```

Comment:
This promise needs to be resolved when the fullscreen status has changed. We can probably use this instead:
```rust
let global = self.global().r();
Promise::Resolve(global, global.get_cx(), HandleValue::undefined())
```